components/Common/LanguageSelector/index.tsx

```diff
@@ -15,6 +16,8 @@ const dropdownStyle = {
 const LanguageSelector = () => {
   const [showDropdown, setShowDropdown] = useState(false);
 
+  const ref = useClickOutside<HTMLDivElement>(() => setShowDropdown(false));
```

Comment:
Since we're passing `() => setShowDropdown(false)` directly from the render function, this will create a new function during each render, causing the `useEffect` inside `useClickOutside` to keep rerunning. You can mitigate the issue by using `useCallback` when you create the render function.

```suggestion
  const dropdownHandler = useCallback(() => setShowDropdown(false), []);
  const ref = useClickOutside<HTMLDivElement>(dropdownHandler);
```